### PR TITLE
Feature/integrate sqlite

### PR DESCRIPTION
**Pull request by sacOO7**
- Added sqlite as a optional dependency
- Updated knex-client file to fit sqlite queries ( should be compatible with pg too)
- Updated knexfile config. for sqlite based on USE_SQLITE flag

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -12,6 +11,7 @@ class DAL {
     config = config || {};
     this.logger = config.logger || console;
     this.knexClient = new KnexClient(config);
+    this.parsers = new DalParser(this.knexClient).getAppliedParsers()
```
Comment (jondubois) [inline src/index.js:14]: Missing semicolon at the end.

File: src/parsers/index.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
+const {
+    applyParserForEach,
+    base64ObjParser,
+    booleanParser,
+    numberParser,
+    removePrivateBlockField,
+    sanitizeTransaction,
+    textToArray
+} = require("./parsers");
```
Comment (jondubois) [inline src/parsers/index.js:9]: Single quote string.

File: src/parsers/index.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+const {accountsTable, transactionsTable, blocksTable, delegatesTable, ballotsTable} = require('../../knex/ldpos-table-schema');
+
+class DalParser {
+
```
Comment (jondubois) [inline src/parsers/index.js:14]: This line spacing before the constructor is inconsistent with other classes.

File: src/parsers/index.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    constructor(knexClient) {
+        this.knexClient = knexClient
+        this.initializeParsers()
```
Comment (jondubois) [inline src/parsers/index.js:17]: Missing semicolons at the end of the lines.

File: src/parsers/index.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                accountsTable.field.nextForgingKeyIndex,
+                accountsTable.field.nextMultisigKeyIndex,
+                accountsTable.field.nextSigKeyIndex,
+                accountsTable.field.requiredSignatureCount,
+            ]
+            parsers.push((account) => numberParser(account, integerFields))
+        }
+
+        return parsers
+    };
```
Comment (jondubois) [inline src/parsers/index.js:38]: Missing semicolons in several places.

File: src/parsers/index.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        ];
+        const textArrayFields = [
+            transactionsTable.field.memberAddresses,
+        ];
+
+        let parsers = [
+            sanitizeTransaction,
+            (txn) => numberParser(txn, bigIntegerFields),
+            (txn) => base64ObjParser(txn, base64Fields),
+            (txn) => textToArray(txn, textArrayFields)
```
Comment (jondubois) [inline src/parsers/index.js:60]: Missing trailing comma at the end. Consistency.

File: src/parsers/index.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        ]
+
+        if (this.knexClient.isSqliteClient()) {
+            const integerFields = [
+                transactionsTable.field.indexInBlock,
+                transactionsTable.field.requiredSignatureCount
```
Comment (jondubois) [inline src/parsers/index.js:66]: Missing comma. Inconsistency.

File: src/parsers/index.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+            ]
+            const integerFields = [
+                blocksTable.field.numberOfTransactions
+            ]
+            parsers.push(
+                (block) => numberParser(block, integerFields),
+                (block) => booleanParser(block, booleanFields)
+            )
+        }
+        return parsers
```
Comment (jondubois) [inline src/parsers/index.js:102]: There are a few places missing trailing commas and semicolons.

File: src/parsers/index.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+    initializeParsers = () => {
+        this.accountTableParsers = this.generateAccountsTableParsers()
+        this.transactionsTableParsers = this.generateTransactionTableParsers()
+        this.blocksTableParsers = this.generateBlocksTableParser()
+        this.delegatesTableParsers = this.generateDelegatesTableParser()
+        this.ballotsTableParsers = this.generateBallotsTableParser()
+    }
+
+    getAppliedParsers = () => {
```
Comment (jondubois) [inline src/parsers/index.js:133]: The name `getAppliedParsers` sounds technical (sounds like a private method). How about something more meaningful like `getRecordParsers`? Or `getResultParsers`?

**Review by jondubois — CHANGES_REQUESTED** [review]
Looks good, mostly small code style inconsistencies.

File: knex/knex-client.js [hunk 10]
```diff
@@ -36,24 +40,29 @@ class KnexClient {
...
+    this.knex = knex(this.knexConfig)
     if (isLocal(environment)) {
       this.knex.on('query', (...args) => {
         if (process.env.KNEX_DEBUG) {
           this.logger.info(args);
         }
       });
+      this.knex.on('query-error', (...args) => {
+          this.logger.error(args);
+      });
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply knex/knex-client.js]: I have removed this one

File: src/parsers/index.js (hunk 9 above)
Comment (sacOO7) [reply src/parsers/index.js]: Updated to recorded parsers

**Pull request merged**